internal/k8s/k8s.go

```diff
@@ -242,27 +242,36 @@ func New(tenant string, cfg Config, db Database, fake bool, log logrus.FieldLogg
 			dynamicClient: dynamicClient,
 		}
 
+		externalResourceInformers := []struct {
+			GroupVersion schema.GroupVersion
+			Resource     string
+			Informer     informers.GenericInformer
+		}{
+			{
+				GroupVersion: kafka_nais_io_v1.GroupVersion,
+				Resource:     "topics",
+				Informer:     infs[cluster].KafkaTopic,
+			},
+			{
+				GroupVersion: aiven_io_v1alpha1.GroupVersion,
+				Resource:     "redis",
+				Informer:     infs[cluster].Redis,
+			},
+			{
+				GroupVersion: aiven_io_v1alpha1.GroupVersion,
+				Resource:     "opensearches",
+				Informer:     infs[cluster].OpenSearch,
+			},
+		}
+
 		if clientSet, ok := clientSet.(*kubernetes.Clientset); ok {
-			resources, err := discovery.NewDiscoveryClient(clientSet.RESTClient()).ServerResourcesForGroupVersion(kafka_nais_io_v1.GroupVersion.String())
-			if err != nil && !strings.Contains(err.Error(), "the server could not find the requested resource") {
-				return nil, fmt.Errorf("get server resources for group version: %w", err)
-			}
-			if err == nil {
-				for _, r := range resources.APIResources {
-					if r.Name == "topics" {
-						infs[cluster].KafkaTopic = dinf.ForResource(kafka_nais_io_v1.GroupVersion.WithResource("topics"))
-					}
-				}
+			if err := externalInformerResources(externalResourceInformers, clientSet, dinf); err != nil {
+				return nil, err
 			}
 		} else if fake {
-			infs[cluster].KafkaTopic = dinf.ForResource(kafka_nais_io_v1.GroupVersion.WithResource("topics"))
-		}
-	}
-
-	if impersonationClientCreator == nil {
-		log.Warnf("impersonation not configured; using default clientSets")
-		impersonationClientCreator = func(ctx context.Context) (map[string]clients, error) {
-			return clientSets, nil
+			for _, externalResourceInformer := range externalResourceInformers {
+				externalResourceInformer.Informer = dinf.ForResource(externalResourceInformer.GroupVersion.WithResource(externalResourceInformer.Resource))
```

Comment:
Vil `externalResourceInformer.Informer` faktisk oppdatere `infs[cluster].` riktig? Ville forventet at f.eks. `infs[cluster].KafkaTopic`  ikke blir oppdatert.

Reply:
Ja det er nok helt riktig at det ikke vill bli satt noe der ja.

Reply:
fungerte lokalt for fake client i hvertfall

Reply:
ja, nu skjønner jeg :D

Reply:
ser riktig ut vil jeg si, testet lit lokalt